enumerables.rb

```diff
@@ -0,0 +1,141 @@
+require 'byebug'
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    if is_a? Array
+      while i <= length - 1
+        yield (self[i])
+        i += 1
+      end
+    elsif is_a? Hash
+      arr = to_a
+      while i <= length - 1
+        yield [arr[i][0], arr[i][1]]
+        i += 1
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    if is_a? Array
+      while i <= length - 1
+        yield [self[i], i]
+        i += 1
+      end
+    elsif is_a? Hash
+      arr = to_a
+      while i <= length - 1
+        yield [arr[i], i]
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a? Hash
+      hashy = {}
+      my_each { |item| hashy[item.to_a[0]] = item.to_a[1] if yield i.to_a[0], i.to_a[1] }
+    else
+      arr = []
+      to_a.my_each { |item| arr << item if yield(item) }
+      arr
+    end
+  end
+
+  def my_all?(data = nil)
+    return my_none?(data) if block_given? && !data.nil?
+
+    arr = to_a
+    return true if arr.empty?
+
+    if block_given?
+      to_a.my_each { |item| return false unless yield(item) }
+    elsif data.nil? && !block_given?
+      to_a.my_each { |item| return false unless item }
+    elsif data.is_a? Class
+      to_a.my_each { |item| return false unless item.is_a? data }
+    elsif data.is_a? Regexp
+      to_a.my_each { |item| return false unless item.to_s.match(data) }
+    end
+    true
+  end
+
+  def my_any?(data = nil)
+    return my_any?(data) if block_given? && !data.nil?
+
+    if block_given?
+      to_a.my_each { |item| return true if yield item }
+      false
+    elsif data.is_a? Regexp
+      to_a.my_each { |item| return true if item.to_s.match(data) }
+    elsif data.is_a? Class
+      to_a.my_each { |item| return true if item.is_a? data }
+    else
+      to_a.my_each { |item| return true if item == data }
+    end
+    false
+  end
```

Comment:
- For my_any? handle the case when no block or argument is given returns true if at least one of the collection is not false or nil

For example:
The built-in method
`[nil, false, nil, false].any?` returns `false`

Your method
`[nil, false, nil, false].my_any?` returns `true`

- It should return false since none of the elements are true.